src/get_address_nbgl.c

```diff
@@ -0,0 +1,101 @@
+#ifdef HAVE_NBGL
+#include "get_address.h"
+#include "utils.h"
+#include "crypto.h"
+#include "menu.h"
+#include "nbgl_use_case.h"
+
+static bool     _generated;
+static uint32_t _account = 0;
+static char     _bip44_path[27]; // max length when 44'/12586'/4294967295'/0/0
+static char     _address[MINA_ADDRESS_LEN];
+
+typedef struct {
+    nbgl_layoutTagValue_t tagValuePair[3];
+    nbgl_layoutTagValueList_t tagValueList;
+    nbgl_pageInfoLongPress_t infoLongPress;
+} TransactionContext_t;
+
+static TransactionContext_t transactionContext;
+
+static uint8_t set_result_get_address(void)
+{
+    uint8_t tx = 0;
+    memmove(G_io_apdu_buffer + tx, _address, sizeof(_address));
+    tx += sizeof(_address);
+    return tx;
+}
+
+static void approve_callback(void)
+{
+    nbgl_useCaseStatus("ADDRESS\nVERIFIED", true, ui_idle);
+}
+
+static void cancel_callback(void)
+{
+    sendResponse(0, false);
+    nbgl_useCaseStatus("Address rejected", false, ui_idle);
+}
+
+static void confirmation_callback(bool confirm) {
+    if (confirm) {
+        sendResponse(set_result_get_address(), true),
+        approve_callback();
+    }
+    else {
+        cancel_callback();
+    }
+}
+
+static void gen_address(void)
+{
+    if (!_generated) {
```

Comment:
I don't get the point of this? Even on Bagl this doesn't seem's to be necessary?